platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -341,6 +355,71 @@ MGL_EXPORT
  */
 @property (nonatomic, readonly) unsigned long long countOfBytesCompleted;
 
+
+#pragma mark - Managing Ambient Cache
+
+/**
+ Sets the maximum ambient cache size in megabytes. The default maximum cache
+ size is 50 MB. To disable ambient caching, set the maximum ambient cache size
+ to `0`. Setting the maximum ambient cache size does not impact the maximum size
+ of offline packs.
+ 
+ While this method does not limit the space available to offline packs,
+ data in offline packs count towards this limit. If the maximum ambient
+ cache size is set to 30 MB and 20 MB of offline packs are downloaded,
+ there may be only 10 MB reserved for the ambient cache.
+ 
+ This method should be called before the map and map style have been loaded.
+ 
+ This method is potentially expensive, as the database will trim cached data
+ in order to prevent the ambient cache from being larger than the
+ specified amount.
+ 
+ @param cacheSize The maximum size in bytes for the ambient cache.
+ @param completion The completion handler to call once the maximum ambient cache size
+ has been set. This handler is executed synchronously on the main queue.
+ */
+
+- (void)setMaximumAmbientCacheSize:(NSUInteger)cacheSize withCompletionHandler:(void (^)(NSError *_Nullable error))completion;
+
+/**
+ Invalidates the ambient cache. This method checks that the tiles in the
+ ambient cache match those from the server. Local tiles that do not match
+ the latest version on the server are updated.
+ 
+ This is more efficient than cleaning the cache because valid local tiles will
+ not be downloaded again.
```

Comment:
I'd say this is not quite accurate: this method clears the entire ambient cache by marking tiles as invalid. Currently there's no way to specify which specific tiles are valid or not; all tiles in the ambient cache are considered invalid and re-downloaded if this method is triggered.

This is more efficient than **deleting** the ambient cache or the offline database. I also recommend referencing the specific APIs for those other two methods in parentheses so it's clear that this is the most preferred method for clearing the ambient cache. 

It might even be worth making that the last line in this section of documentation:

"This is the recommended method for clearing the ambient cache. It's the most efficient and least computationally expensive method."

Reply:
Okay, sounds good. I was going off the core documentation. It sounds like the behavior doesn't match this? https://github.com/mapbox/mapbox-gl-native/blob/68e3cbacbc35ae6cd6c1e0fe2ab59ed262c32786/include/mbgl/storage/default_file_source.hpp#L196-L200

Reply:
Ahh, thanks. I'll try to sync with @tmpsantos tomorrow to clear things up. cc @zugaldia